orchagent/txmonorch.cpp

```diff
@@ -0,0 +1,313 @@
+                periodToSet = to_uint<uint32_t>(fvValue(i));
+
+                needStart |= (periodToSet != m_pollPeriod);
+                SWSS_LOG_INFO("TX_ERR handle cfg update period new %d\n", periodToSet);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown field type %s\n", fvField(i).c_str());
+                return -1;
+            }
+        }
+        catch (...) {
+            SWSS_LOG_ERROR("Failed to handle period update\n");
+        }
+    }
+
+    if (needStart)
+    {
+        startTimer(periodToSet);
+        SWSS_LOG_INFO("TX_ERR poll timer restarted with interval %d\n", periodToSet);
+    }
+
+    return 0;
+}
+
+int TxMonOrch::handleThresholdUpdate(const string &port, const vector<FieldValueTuple>& data, bool clear)
+{
+    SWSS_LOG_ENTER();
+
+    try {
+        if (clear)
+        {
+            //attention, for clear, no data is empty
+            //tesThreshold(m_PortsTxErrStat[port]) = 0;
+            m_PortsTxErrStat.erase(port);
+            m_TxErrorTable.del(port);
+            m_stateTxErrorTable.del(port);
+            //todo, remove data from state_db and appl_db??
+            SWSS_LOG_INFO("TX_ERR threshold cleared for port %s\n", port.c_str());
+        }
+        else
+        {
+            for (auto i : data)
+            {
+                if (TXMONORCH_FIELD_CFG_THRESHOLD == fvField(i))
+                {
+                    TxErrorStatistics &tes = m_PortsTxErrStat[port];
+                    if (tesPortId(tes) == 0/*invalid id??*/)
+                    {
+                        //the first time this port is configured
+                        Port saiport;
+                        //what if port doesn't stand for a valid port?
+                        //that is, getPort returns false?
+                        //what if the interface is removed with threshold configured?
+                        if (gPortsOrch->getPort(port, saiport))
+                        {
+                            tesPortId(tes) = saiport.m_port_id;
+                        }
+                        tesState(tes) = TXMONORCH_PORT_STATE_UNKNOWN;
+                    }
+                    tesThreshold(tes) = to_uint<uint64_t>(fvValue(i));
+                    SWSS_LOG_INFO("TX_ERR threshold reset to %ld for port %s\n",
+                                  tesThreshold(tes), port.c_str());
+                }
+                else
+                {
+                    SWSS_LOG_ERROR("Unknown field type %s when handle threshold for %s\n",
+                                   fvField(i).c_str(), port.c_str());
+                    return -1;
+                }
+            }
+        }
+    }
+    catch (...) {
+        SWSS_LOG_ERROR("Fail to startTimer handle periodic update\n");
+    }
+
+    return 0;
+}
+
+/*handle configuration update*/
+void TxMonOrch::doTask(Consumer& consumer)
+{
+    int rc = 0;
+
+    SWSS_LOG_ENTER();
+    SWSS_LOG_INFO("TxMonOrch doTask consumer\n");
```

Comment:
the swss_log_enter pretty much does the same print only it is in DEBUG mode and not INFO.
for your consideration